### PR TITLE
Global input parse

### PR DESCRIPTION
**Pull request by tiosgz**
This moves input parsing (yes|no, trim) into functions so we don't have to make one change on many places

### CODE REVIEW

**Comment by tiosgz** [comment]
What I was wondering about:
* do we really need to `echo -e` user input? (see how it now looks like when you trim something)
* we can use https://stackoverflow.com/a/3352015 or https://stackoverflow.com/a/369795 for trimming newlines (and it's w/o external tools), but then we would need to use `sed` anyway if we want to strip also whitespace at the beginning/end of an "inner" line. What's the "correct" solution for you?
* do you want `>>>` at the beginning of every prompt? I removed most of them while changing to `resolve_boolean`

File: stup [hunk 1]
```diff
@@ -163,17 +163,14 @@ execute_show()
       echo ""
     else
       print_warning "No entries found for $(display_date $AT) in category $CATEGORY."
-      echo -e "Do you want to retrieve the last notes added in this category with stup?"
-      read -e -p "Type y|Y|yes|Yes or anything else to exit: " search_last_entries
+      read -e -p "Do you want to retrieve the last notes added in this category with stup? " search_last_entries
```
Comment (iridakos) [inline stup:166]: I think it would be nice to provide available options - not all but only `(y/n)` as hint of what we expect from the users to type.

**Comment by iridakos** [comment]
First of all: thank you very much for your contributions!

> * do we really need to `echo -e` user input? (see how it now looks like when you trim something)

:+1: we don't need that

> * we can use https://stackoverflow.com/a/3352015 or https://stackoverflow.com/a/369795 for trimming newlines (and it's w/o external tools), but then we would need to use `sed` anyway if we want to strip also whitespace at the beginning/end of an "inner" line. What's the "correct" solution for you?

Pick your favorite.

> * do you want `>>>` at the beginning of every prompt? I removed most of them while changing to `resolve_boolean`

Nope, you can either remove the rest or else I can do it too.

**Comment by tiosgz** [comment]
I realised that the second StackOverflow answer did exactly what the very first solution (removed all whitespace), and the first one doesn't allow changing the `#!` to `zsh`, so I used https://unix.stackexchange.com/a/476504 instead.

Also, please keep this until #37 is merged, I have some changes for the `ynaq` there.

**Comment by iridakos** [comment]
Sure! Thank you :)

**Comment by iridakos** [comment]
@tiosgz Let me know whenever this PR is ready in order to merge it. #37 is already merged, you should be good to go.

**Comment by tiosgz** [comment]
I think it's ready. I can remove the TODO and add `resolve_ynqa` right now if you want.

**Comment by iridakos** [comment]
@tiosgz I agree with your comment. We can put it in another function once we see that there's a need to re-use it.

Thank you very much for your excellent contributions :pray:

**Pull request merged**